sigar_windows.go

```diff
@@ -117,8 +118,70 @@ func (self *Cpu) populateFromPdh(values []uint64) {
 	self.Irq = values[3]
 }
 
+// Only get the first mount point for each volume
+func getVolumePathName(volumeName []uint16) (string, error) {
+	charsLen := C.DWORD(C.MAX_PATH)
+	var chars []uint16
+	for {
+		chars = make([]uint16, charsLen)
+		if C.GetVolumePathNamesForVolumeNameW((*C.WCHAR)(&volumeName[0]), (*C.WCHAR)(&chars[0]), C.MAX_PATH, &charsLen) == C.TRUE {
+			break
+		}
+		errno := syscall.GetLastError()
+		err, ok := errno.(syscall.Errno)
+		if !ok || err != C.ERROR_MORE_DATA {
+			return "", errno
+		}
+	}
+	volumePaths := syscall.UTF16ToString(chars)
+	return volumePaths, nil
+}
+
+// Get a list of local filesystems
+// Does not apply to SMB volumes
 func (self *FileSystemList) Get() error {
-	return notImplemented()
+	bufSize := C.DWORD(C.MAX_PATH)
+	buf := make([]uint16, bufSize)
+
+	capacity := len(self.List)
+	if capacity == 0 {
+		capacity = 4
+	}
+	self.List = make([]FileSystem, capacity)
+
+	// Get a handle to iterate over all the attached disks
+	diskHandle := C.FindFirstVolumeW((*C.WCHAR)(&buf[0]), bufSize)
+	if diskHandle == C.HANDLE(uintptr(0xFF)) {
+		return fmt.Errorf("Got bad handle when calling FindFirstVolume")
+	}
+
+	for {
+		volume := syscall.UTF16ToString(buf)
+
+		// Get the first mount point for each volume
+		volumeMount, err := getVolumePathName(buf)
+		if err != nil {
+			return fmt.Errorf("Error getting volume mountpoints: %v", err)
+		}
+
+		self.List = append(self.List, FileSystem{
+			DirName: volumeMount,
+			DevName: volume,
+		})
+
+		if C.FindNextVolumeW(diskHandle, (*C.WCHAR)(&buf[0]), bufSize) == C.FALSE {
```

Comment:
I had to reread this function a bunch of times, and I really didn't understand what was going on until I read through the docs for `FindFirstVolume` and `FindNextVolume`.  I think these are kind of baroque and made this function hard to follow.  Have we thought about wrapping them in some struct that we can use some what like an iterator?